### PR TITLE
Get model group API

### PR DESCRIPTION
**Pull request by rbhavna**
### Description
Get model group API using model_group_id
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetRequest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+@ToString
+public class MLModelGroupGetRequest extends ActionRequest {
+
+    String modelGroupId;
+    boolean returnContent;
+
+    @Builder
+    public MLModelGroupGetRequest(String modelGroupId, boolean returnContent) {
+        this.modelGroupId = modelGroupId;
+        this.returnContent = returnContent;
```
Comment (ylwu-amzn) [inline common/src/main/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetRequest.java:38]: Why add this parameter? Can you add some example request/response to the description part?

File: plugin/src/main/java/org/opensearch/ml/action/model_group/GetModelGroupTransportAction.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                if (r != null && r.isExists()) {
+                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+
+                        MLModelGroup mlModelGroup = MLModelGroup.parse(parser);
+                        modelAccessControlHelper.validateModelGroupAccess(user, modelGroupId, client, ActionListener.wrap(access -> {
+                            if (!access) {
+                                wrappedListener
+                                    .onFailure(
+                                        new MLValidationException(
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/action/model_group/GetModelGroupTransportAction.java:90]: Let's add this [type of exception](https://github.com/opensearch-project/ml-commons/blob/feature/agent_framework_dev/plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java#L112-L116) as this shouldn't be a 500 level error.

File: common/src/main/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetRequest.java (hunk 1 above)
Comment (rbhavna) [reply common/src/main/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetRequest.java:38]: We might not need this. This is mostly related to model content. Removing this for model group API

File: common/src/test/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetResponseTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+
+public class MLModelGroupGetResponseTest {
+
+    MLModelGroup mlModelGroup;
+
+    @Before
+    public void setUp() {
+        mlModelGroup = MLModelGroup.builder()
```
[the remark this answers is not in the document]
Comment (rbhavna) [reply common/src/test/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetResponseTest.java:34]: To test such complex cases, we might need to add them in IT security tests. In this class we can only test two cases where user has access or no access to model group. Will add them in security tests

File: plugin/src/main/java/org/opensearch/ml/action/model_group/GetModelGroupTransportAction.java (hunk 2 above)
Comment (rbhavna) [reply plugin/src/main/java/org/opensearch/ml/action/model_group/GetModelGroupTransportAction.java:90]: Sure will do

File: common/src/test/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetResponseTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                "\"access\":\"public\"}",
+                jsonStr);
+    }
+
+    @Test
+    public void fromActionResponseWithMLModelGetResponse_Success() {
+        MLModelGroupGetResponse mlModelGroupGetResponse = MLModelGroupGetResponse.builder().mlModelGroup(mlModelGroup).build();
+        MLModelGroupGetResponse mlModelGroupGetResponseFromActionResponse = MLModelGroupGetResponse.fromActionResponse(mlModelGroupGetResponse);
+        assertSame(mlModelGroupGetResponse, mlModelGroupGetResponseFromActionResponse);
+        assertEquals(mlModelGroupGetResponse.mlModelGroup, mlModelGroupGetResponseFromActionResponse.mlModelGroup);
```
Comment (dhrubo-os) [inline common/src/test/java/org/opensearch/ml/common/transport/model_group/MLModelGroupGetResponseTest.java:73]: We are checking references here, is that the goal?

**Review by dhrubo-os — APPROVED** [review]

**Comment by dhrubo-os** [comment]
apply spotless.

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**